lib/msf/ui/console/command_dispatcher/common.rb

```diff
@@ -158,6 +158,31 @@ def index_from_list(list, index, &block)
     yield list[idx]
   end
 
+  # Trims starting `.`, `./` `/`, `+path_head+/`, & `/+path_head+/` from +path+. Also trims trailing `.+extension+`
+  # from +path+, and any possible combination of misspellings of +extension+.
+  #
+  # @param path [String] The path to be trimmed
+  # @param path_head [String] The top-level directory that should be removed from the path
+  # @param extension [String] File extension to be trimmed from +path+. `.` is automatically included. Defaults to `rb`.
+  # @return [String] Altered +path+. Will return unaltered +path+ if regex constructed with +path_head+ & +path+ is not detected
+  def trim_path(path, path_head, extension="rb")
+    regexp = %r{
+        (
```

Comment:
You're using a non-capturing group in this capturing group but not using the match. Were you going to do something with it?

Reply:
Yeah the non-capturing group shouldn't be in there, good 👁. Weird thing is the path_head still gets captured with it in the regex, all taking it out does is add an additional match which is a little odd

Reply:
Any idea what's causing this @wvu-r7?

Reply:
@adamgalway-r7 What are your thoughts on adding a few unit tests to fully test this method? 🔬

Reply:
Doing it now 👍

Reply:
I mean, if you're not using any matches, the regex can probably use non-capturing groups. But it does add complexity to the expression. Either way, don't care. So long as it's consistent. :)

Reply:
Tests are added in and consistency is (hopefully) guaranteed 🤞